backend/src/contaxy/api/endpoints/auth.py

```diff
@@ -172,6 +172,87 @@ def list_api_tokens(
     )
 
 
+@router.get(
+    "/auth/permissions",
+    summary="Lists roles for given resource.",
+    operation_id=CoreOperations.GET_RESOURCE_PERMISSIONS.value,
+    response_model=List[str],
+)
+def get_resource_permissions(
+    resource_name: str,
+    resolve_roles: bool = True,
+    use_cache: bool = True,
+    token: str = Depends(get_api_token),
+    component_manager: ComponentManager = Depends(get_component_manager),
+) -> Any:
+    """Lists the roles for a given resource, admin access for that resource is required."""
+    component_manager.verify_access(token, resource_name, AccessLevel.ADMIN)
+
+    return component_manager.get_auth_manager().list_permissions(
+        resource_name, resolve_roles, use_cache
+    )
+
+
+@router.post(
+    "/auth/permissions",
+    summary="Sets Permission for resource.",
+    operation_id=CoreOperations.SET_RESOURCE_PERMISSIONS.value,
+    status_code=status.HTTP_204_NO_CONTENT,
+)
+def set_resource_permissions(
+    resource_name: str,
+    permission: str,
+    token: str = Depends(get_api_token),
+    component_manager: ComponentManager = Depends(get_component_manager),
+) -> Any:
+    """Sets Permission for resource, admin access for that resource is required."""
+    component_manager.verify_access(token, resource_name, AccessLevel.ADMIN)
```

Comment:
This would allow a user (that by default has access to its own user resource) to add any permission to itself! This would of course be a security issue. We could think about allowing a user to give other users access to resources it owns. But that might also cause issues and requires more thought. For now, I would only allow super admins to use this endpoint, so they can fix permission related issues without the need to do it directly in the DB. 
To do that, you could modify the check like this: `component_manager.verify_access(token, "/auth/permissions", AccessLevel.ADMIN)`. A normal user does not have the permission for "/auth/permissions" and cannot use the endpoint. The super admin has the "*#admin" permission and can therefore access all endpoints.
The same issue exists for delete and list resources with permission endpoints below. They should also not be accessible by all users.